frontend/server/src/main/java/org/pytorch/serve/http/TokenAuthorizationHandler.java

```diff
@@ -84,20 +108,208 @@ public static void setupTokenClass() {
     private void checkTokenAuthorization(FullHttpRequest req, String type) throws ModelException {
 
         try {
-            Method method =
-                    tokenClass.getMethod(
-                            "checkTokenAuthorization",
-                            io.netty.handler.codec.http.FullHttpRequest.class,
-                            String.class);
-            boolean result = (boolean) (method.invoke(tokenObject, req, type));
+            boolean result = tokenClass.checkTokenAuthorization(req, type);
             if (!result) {
                 throw new InvalidKeyException(
                         "Token Authentication failed. Token either incorrect, expired, or not provided correctly");
             }
-        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
-            e.printStackTrace();
+        } catch (Exception e) {
             throw new InvalidKeyException(
                     "Token Authentication failed. Token either incorrect, expired, or not provided correctly");
         }
     }
 }
+
+class Token {
+    private static String apiKey;
+    private static String managementKey;
+    private static String inferenceKey;
+    private static Instant managementExpirationTimeMinutes;
+    private static Instant inferenceExpirationTimeMinutes;
+    private static Double timeToExpirationMinutes;
+    private SecureRandom secureRandom = new SecureRandom();
+    private Base64.Encoder baseEncoder = Base64.getUrlEncoder();
+    private String fileName = "key_file.json";
+    private String filePath = "";
+
+    public String updateKeyFile(FullHttpRequest req) throws IOException {
```

Comment:
Lets not pull HttpRequest into this function but instead call generateKeyFile directly and parse request outside of Token. This class is highly critical, so we should keep it as lean as possible.